Dip/DipTests/ComponentScopeTests.swift

```diff
@@ -90,14 +90,14 @@ class ComponentScopeTests: XCTestCase {
 
   func testThatItReusesInstanceInObjectGraphScopeDuringResolve() {
     //given
-    container.register(.ObjectGraph) { [unowned container] in Client(server: container.resolve()) as Client }
+    container.register(.ObjectGraph) { [unowned container] in Client(server: try! container.resolve()) as Client }
     
     container.register(.ObjectGraph) { Server() as Server }.resolveDependencies { container, server in
```

Comment:
Not related to this PR I know, but now that I read that again I'm wondering: shouldn't we use `[unowned container]` (or at least `[weak container]`) here? It seems to me that `container` strongly reference `resolveDependenciesBlock` which itself strongly references `container`, hence a strong reference cycle, right?